custom/motioncontrollers/MotionController.java

```diff
@@ -217,4 +282,41 @@ public void disable() {
 	public boolean onTarget() {
 		return Math.abs(getError()) <= absoluteTolerance;
 	}
+	
+	/**
+	 * Check if the motion controller has generated
+	 * an exception within the TimerTask. If there is
+	 * not an exception, the function returns null.
+	 *
+	 * @return the exception (probably null)
+	 */
+	public Exception checkException() {
+		return mcException;
+	}
+	
+	/**
+	 * The thread in which the output is updated with the
+	 * results of the motion controller calculation.
+	 *
+	 */
+	protected class MotionControllerTask extends TimerTask {
+		@Override
+		public void run() {
+			try {
+				double value = getSafely(); // Always calculate MC output
+				if (justReset) {
+					synchronized (justReset) {
+						justReset = false;
+					}
```

Comment:
Checking the value outside of a synchronized context, then SETTING the value in its own synchronized context????????